src/index.ts

```diff
@@ -12,6 +12,68 @@ export interface AccessControlOptions {
   preflightContinue?: boolean
 }
 
+const isIterable = (obj: unknown): obj is Iterable<unknown> => typeof obj[Symbol.iterator] === 'function'
+
+const getOriginHeaderHandler = (origin: unknown): ((req: Request, res: Response) => void) => {
+  const fail = () => {
+    throw new TypeError('No other objects allowed. Allowed types is array of strings or RegExp')
+  }
+
```

Comment:
this can be moved outside of the function to the upper level

Reply:
But it wouldn't make sense out of context - I.e. it would become possible to call `fail` from anywhere, which shouldn't be possible as its message is context specific.

Reply:
If you would like me to proceed with this change regardless, please let me know

Reply:
if it "would" be possible doesn't mean it would be done. no need to declare pure functions within another function.

Reply:
I prefer methods that make mistakes **impossible** rather than **improbable**.
The need here is clearly evident. 
Please let me know if you would like me to proceed with this change regardless.

Reply:
I would prefer you to proceed and move it out to the global scope. this package is really tiny and a chance for mistakes like this is miniscule. Especially since I'm the main maintainer anyway.